doc/tutorial.md

```diff
@@ -1778,6 +1778,40 @@ It is often a mistake to silently ignore an exception, so if the future we're ig
 
 The ```handle_connection()``` function itself is straightforward --- it repeatedly calls ```read()``` read on the input stream, to receive a ```temporary_buffer``` with some data, and then moves this temporary buffer into a ```write()``` call on the output stream. The buffer will eventually be freed, automatically, when the ```write()``` is done with it. When ```read()``` eventually returns an empty buffer signifying the end of input, we stop ```repeat```'s iteration by returning a ```stop_iteration::yes```.
 
+Re-written using C++20's coroutines, the above becomes this:
+
```

Comment:
IMO it should replace the original. We encourage coroutines now (except in one case - when the function usually resolves with a ready future and it is very performance sensitive). The original is very outdated, no one should use keep_doing().

Reply:
I think eventually we need to go over the tutorial and reorganize/rewrite it in a way that gives more explanations and examples using coroutines - and only later in a separate section introduces the finer points of continuations, how to use them, when to use them, and so on. In the continuations section we'll need some examples, including of keep_doing, but it doesn't need to be this specific server.

Reply:
Given the following goals:
- A: Remove the continuation-based version of the echo server and adjust the explanation to match the coroutine-based version.
- B: Convert the whole tutorial from continuations to coroutines.

I guess we have the following options for this PR here:
- Discard this PR (close unmerged).
- Merge it as it is (keep A and B for later)
- I do A in this PR.

(Doing B currently feels way above my understanding/skills.)